lib_i2s/src/i2s_frame_master_impl.h

```diff
@@ -69,24 +70,25 @@ static i2s_restart_t i2s_frame_ratio_n(client i2s_frame_callback_if i2s_i,
 
 #pragma loop unroll
     for (size_t i=0, idx=0; i<num_out; i++, idx+=2){
-        p_dout[i] @ (1 + offset) <: bitrev(out_samps[idx]);
+        partout_timed(p_dout[i], num_data_bits, bitrev(out_samps[idx]), (1 + offset));
     }
 
-    p_lrclk @ 1 <: lr_mask;
+    partout_timed(p_lrclk, num_data_bits, lr_mask, 1);
```

Comment:
My OCD doesn't like the fact that these changes modify the behaviour for 1-bit ports (where you don't really want a partout since it add's extra instructions). num_data_bits should be a const, therefore an If(data_bits  == 1) would (should) be optimised away by the compiler.  To improve readability (rather than if's all over the place you could and your online version of partout (static inline) within this file that includes the if.